assets/js/translator.js

```diff
@@ -646,8 +653,41 @@ function populateTranslationList() {
             }
         }
 
-        srcLangs = srcLangs.sort(compareLangCodes);
-        dstLangs = dstLangs.sort(function (a, b) {
+        var langsOnly = [];
+        var variantsOnly = [];
+
+        for(i = 0; i < srcLangs.length; i++) {
+            if(srcLangs[i].indexOf('_') !== -1) {
+                variantsOnly.push(srcLangs[i]);
+            }
+            else {
+                langsOnly.push(srcLangs[i]);
+            }
+        }
+
+        srcLangs = langsOnly.sort(compareLangCodes);
+        for(i = 0; i < variantsOnly.length; i++) {
```

Comment:
While the double for-loop works, I think there's a more efficient way to do it:
1. sort `langsOnly`
2. sort `variantsOnly`
3. have `var langHead = 0` and `var variantHead = 0`
4. in a single for-loop, compare the lang at `langsOnly[langHead]` with the variant at `variantsOnly[variantHead]`, and depending on the comparison, either `langHead++` or `variantHead++`; simultaneously build the entire `srcLangs` list

(If that's not clear, ask on IRC :smiley:)